python/lsst/daf/butler/core/connectionString.py

```diff
@@ -80,12 +80,20 @@ def fromConfig(cls, registryConfig):
             if getattr(conStr, key) is None:
                 setattr(conStr, key, regConf.get(key))
 
-        # sqlite with users and passwords not supported
-        if None in (conStr.username, conStr.password) and "sqlite" not in conStr.drivername:
+        # allow other mechanisms to insert username and password by not forcing
+        # the credentials to exist, always re-raise only the case where
+        # credentials# file exists but is incorrect permissions, for safety
+        try:
             dbAuth = DbAuth(DB_AUTH_PATH, DB_AUTH_ENVVAR)
-            auth = dbAuth.getAuth(conStr.drivername, conStr.username, conStr.host,
-                                  conStr.port, conStr.database)
-            conStr.username = auth[0]
-            conStr.password = auth[1]
+        except DbAuthError as e:
+            if 'permissions' in e.args[0]:
```

Comment:
Don't rely on parsing the error message in an attempt to work out what type of problem this was. What you need to do is raise a different exception for permissions problem. You want to be able to code this as something like:
```python
try:
    dbAuth = DbAuth(...)
except DBAuthFileNotFoundError:
   pass
else:
   ...
```
and have the constructor throw `DbAuthPermissionsError` that would not be caught. They could both inherit from `DbAuthError` but you'd only be ignoring the lack of db auth file rather than ignoring everything and trying to parse the error.

Reply:
Yeah, I was pretty sure this would get a comment but wasn't sure whether a `DbAuthPermissionsError` is wanted ot should I just raise a regular python `PermissionError` (which isn't there exactly for this thing, but matched the closest to what I wanted).

Reply:
DbAuthPermissionError could inherit from PermissionError and from DbAuthError depending on how you want people to be catching things.